From 4e5fa5c7b82ae54882908ff418fcb0dd3cd288a9 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:23:13 -0400
Subject: [tools] perf: Update a stale comment

Message-id: <1408462094-14194-25-git-send-email-jolsa@redhat.com>
Patchwork-id: 88089
O-Subject: [PATCH RHEL7.1 BZ1131394 024/325] perf: Update a stale comment
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 394570b7939e1262f39373866166d8ee0a506e88
Author: Peter Zijlstra <peterz@infradead.org>
Date: Thu Oct 31 17:41:23 2013 +0100
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/kernel/events/ring_buffer.c b/kernel/events/ring_buffer.c
index c52a32f..e8b168a 100644
--- a/kernel/events/ring_buffer.c
+++ b/kernel/events/ring_buffer.c
@@ -85,8 +85,8 @@ again:
  rb->user_page->data_head = head;
 
  /*
-  * Now check if we missed an update, rely on the (compiler)
-  * barrier in atomic_dec_and_test() to re-read rb->head.
+  * Now check if we missed an update -- rely on previous implied
+  * compiler barriers to force a re-read.
   */
  if (unlikely(head != local_read(&rb->head))) {
   local_inc(&rb->nest);
-- 
1.7.1